src/jaxsim/parsers/kinematic_graph.py

```diff
@@ -427,19 +496,31 @@ def reduce(self, considered_joints: List[str]) -> KinematicGraph:
 
         # Create the reduced graph data. We pass the full list of links so that those
         # that are not part of the graph will be returned as frames.
-        reduced_root_node, reduced_joints, reduced_frames, unconnected_joints = (
-            KinematicGraph.create_graph(
-                links=list(full_graph_links_dict.values()),
-                joints=[joints_dict[joint_name] for joint_name in considered_joints],
-                root_link_name=full_graph.root.name,
-            )
+        (
+            reduced_root_node,
+            reduced_joints,
+            reduced_frames,
+            unconnected_links,
+            unconnected_joints,
+            unconnected_frames,
+        ) = KinematicGraph._create_graph(
+            links=list(full_graph_links_dict.values()),
+            joints=[joints_dict[joint_name] for joint_name in considered_joints],
+            root_link_name=full_graph.root.name,
+        )
+
+        assert set(f.name for f in self.frames).isdisjoint(
+            set(f.name for f in unconnected_frames + reduced_frames)
         )
 
-        # Create the reduced graph
+        for link in unconnected_links:
+            logging.debug(msg=f"Link '{link.name}' is unconnected and became a frame")
```

Comment:
Definitely not an issue, but I believe that if we use an `f-string` here, Python will try to evaluate this even when the logging level is above `debug`
```suggestion
            logging.debug(msg="Link '{}' is unconnected and became a frame".format(link.name))
```

Reply:
Mmh I think that the string is evaluated regardless since it's not controlled by the logger. It's only the redirection to stdout that is filtered by the logger, all the processing should happen in any case (never checked, just my intuition).

The situation would be different if the fmt processing would be finalized internally like it happens in jax debug messages (for other reasons, in that case).

Reply:
In this case it's definitely not crucial as we are not evaluating anything computationally expensive, so you can ignore this comment. See https://docs.astral.sh/ruff/rules/logging-f-string/ for reference

Reply:
Good to know, thanks for the reference. Btw, what you aimed is not currently possible with our wrapper class of the python logger. The evaluation can be deferred by using the plain logger as:

```python
logger.debug(msg="My string printing {something}", something=something)
```

However, out logger only exposes `msg` and not the optional args /  kwargs.